src/Hangfire.Mongo/PersistentJobQueue/Mongo/MongoJobQueueMonitoringApi.cs

```diff
@@ -32,16 +32,13 @@ public IEnumerable<string> GetQueues()
 
         public IEnumerable<int> GetEnqueuedJobIds(string queue, int @from, int perPage)
         {
-            int start = @from + 1;
-            int end = from + perPage;
-
             return AsyncHelper.RunSync(() => _connection.JobQueue.Find(
                         Builders<JobQueueDto>.Filter.Eq(_ => _.Queue, queue) &
                         Builders<JobQueueDto>.Filter.Eq(_ => _.FetchedAt, null)
-                    ).ToListAsync())
-                .Select((data, i) => new { Index = i + 1, Data = data })
-                .Where(_ => (_.Index >= start) && (_.Index <= end))
-                .Select(x => x.Data)
+                    )
+                    .Skip(@from)
+                    .Limit(perPage)
+                    .ToListAsync())
```

Comment:
Nice one!
I missed this file in my previous pass of doing the same thing.  Good catch.